docs/DeveloperGuide.adoc

```diff
@@ -218,29 +228,431 @@ image::CommitActivityDiagram.png[]
 
 ===== Aspect: How undo & redo executes
 
-* **Alternative 1 (current choice):** Saves the entire address book.
+* **Alternative 1 (current choice):** Saves the entire billboard.
 ** Pros: Easy to implement.
 ** Cons: May have performance issues in terms of memory usage.
 * **Alternative 2:** Individual command knows how to undo/redo by itself.
-** Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
+** Pros: Will use less memory (e.g. for `delete`, just save the expense being deleted).
 ** Cons: We must ensure that the implementation of each individual command are correct.
 
 ===== Aspect: Data structure to support the undo/redo commands
 
-* **Alternative 1 (current choice):** Use a list to store the history of address book states.
+* **Alternative 1 (current choice):** Use a list to store the history of billboard states.
 ** Pros: Easy for new Computer Science student undergraduates to understand, who are likely to be the new incoming developers of our project.
-** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `HistoryManager` and `VersionedAddressBook`.
+** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `Model` and `VersionedBillboard`.
 * **Alternative 2:** Use `HistoryManager` for undo/redo
 ** Pros: We do not need to maintain a separate list, and just reuse what is already in the codebase.
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
-// tag::dataencryption[]
-=== [Proposed] Data Encryption
+// tag::archive[]
+=== Archive
+==== Implementation
+===== Modelling Archive
+The archive feature supports the following actions:
+
+* Creating an archive
+* Added an expense to an archive
+* Reverting/"unarchiving" an archived expense
+* Deleting an archived expense
+* Displaying the list of expense of a particular archive
+* Listing all existing archive names
+
+
+These actions are facilitated by the `ArchiveWrapper` and `Archive` classes:
+
+*  `Archive` extends from `ExpenseList` in order to encapsulate an archive name and a list of expenses together as an archive.
+* `ArchiveWrapper` manages all existing archives and hashes each `Archive` object to its archive name.
+
+Similar to `Billboard`, which manages the list of current non-archived expenses,
+`ArchiveWrapper` is used in `ModelManager` and its respective operations are called to access and manipulate archive expenses when an archive command is entered. +
+Such operations include:
+
+* `ArchiveWrapper#AddArchive(Archive)` - Adds the given archive to the current map of archive objects.
```

Comment:
This information can be found in Java doc. We feel that it's a bit redundant to leave them here.